packages/api/src/routes/devices/internal-user.ts

```diff
@@ -72,4 +67,50 @@ router.post(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * POST /internal/user/resubscribe-fitbit-webhooks
+ *
+ * Finds all existing users that are connected to Fitbit and recreates their webhook subscriptions.
+ *
+ * @param req.query.cxId - (Optional) the customer/account's ID.
+ * @return Count of users processed and count of users with disconnected providers.
+ */
+router.post(
+  "/resubscribe-fitbit-webhooks",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").optional();
+    const connectedUsers = await getAllConnectedUsers(cxId);
+
+    let usersAffected = 0;
+    for (const connectedUser of connectedUsers) {
+      const providers = connectedUser.providerMap ? Object.keys(connectedUser.providerMap) : [];
+      for (const providerStr of providers) {
+        if (providerStr === ProviderSource.fitbit) {
+          const providerOAuth2Type = providerOAuth2OptionsSchema.safeParse(providerStr);
+          if (providerOAuth2Type.success) {
+            try {
+              const fitbitToken = connectedUser.providerMap?.fitbit?.token;
+              if (fitbitToken) {
+                await Constants.PROVIDER_OAUTH2_MAP[ProviderSource.fitbit].postAuth?.(
+                  fitbitToken,
+                  connectedUser
+                );
+                usersAffected++;
+              }
+            } catch (err) {
+              console.log(
+                `Failed to add webhook subscriptions through the internal user route. User: ${connectedUser}, Provider: ${providerStr}, Error: ${err}.`
+              );
+            }
+          }
+        }
+      }
+    }
+    return res.status(status.OK).json({
+      usersProcessed: connectedUsers.length,
+      usersAffected,
+    });
```

Comment:
maybe we can also add a property with the count of errors? so we know if we should investigate?